hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/scale/ITestS3AHugeFileUploadSinglePut.java

```diff
@@ -42,14 +51,23 @@ protected Configuration createScaleConfiguration() {
     configuration.setLong(MULTIPART_SIZE, 53687091200L);
     configuration.setInt(KEY_TEST_TIMEOUT, 36000);
     configuration.setInt(IO_CHUNK_BUFFER_SIZE, 655360);
-    configuration.set("fs.s3a.connection.request.timeout", "1h");
+    configuration.set(REQUEST_TIMEOUT, "1h");
+    fileSize = getTestPropertyBytes(configuration, KEY_HUGE_FILESIZE,
+        DEFAULT_HUGE_FILESIZE);
     return configuration;
   }
 
   @Test
   public void uploadFileSinglePut() throws IOException {
     LOG.info("Creating file with size : {}", fileSize);
-    ContractTestUtils.createAndVerifyFile(getFileSystem(),
-        getTestPath(), fileSize );
+    S3AFileSystem fs = getFileSystem();
+    ContractTestUtils.createAndVerifyFile(fs,
+        getTestPath(), fileSize);
+    //No more than three put requests should be made during the upload of the file
+    //First one being the creation of test/ directory marker
+    //Second being the creation of the file with tests3ascale/<file-name>
+    //Third being the creation of directory marker tests3ascale/ on the file delete
+    assertEquals(3L,
```

Comment:
use `IOStatisticAssertions` here; it generates AssertJ assertion chains from lookups with automatic generation of error text.

```java
assertThatStatisticCounter(fs.getIOStatistics(), OBJECT_PUT_REQUESTS.getSymbol())
  .isEqualTo(3);
```